From 270e0ea666ab6979e545ff5d4786e495febe7dac Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Mon, 19 Jan 2015 11:31:17 +0100
Subject: [PATCH] Remove pkcs11_startup from the initscript

Signed-off-by: Petr Lautrbach <plautrba@redhat.com>
---
 misc/pkcsslotd.in | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/misc/pkcsslotd.in b/misc/pkcsslotd.in
index 5c2c0c7..81cad12 100644
--- a/misc/pkcsslotd.in
+++ b/misc/pkcsslotd.in
@@ -15,19 +15,14 @@
 PIDFILE=/var/run/pkcsslotd.pid
 LOCKFILE=/var/lock/subsys/pkcsslotd
 SLOTDBIN=@sbindir@/pkcsslotd
-CONFSTART=@sbindir@/pkcs11_startup
 
 
 start() {
 	[ -x $SLOTDBIN ] || exit 5
-	[ -x $CONFSTART ] || exit 5
 
  	echo -n $"Starting pkcsslotd: "
 
-        # Generate the configuration information
-        $CONFSTART
-
-        daemon $SLOTDBIN
+	daemon $SLOTDBIN
 
 	RETVAL=$?
 	echo
-- 
2.2.1

